src/components/data-display/tables/DomainsTable.tsx

```diff
@@ -0,0 +1,467 @@
+  status: string | number;
+  action: ReactNode;
+} & Record<string, any>;
+
+const columnHelper = createColumnHelper<TableData>();
+
+function filterTableData(filter: string, data: TableData[]): TableData[] {
+  const results: TableData[] = [];
+
+  data.forEach((d) => {
+    let matchFound = false;
+
+    Object.entries(d).forEach(([, v]) => {
+      if (typeof v === 'object' && v !== null) {
+        // Recurse into nested objects
+        const nestedResults = filterTableData(filter, [v]);
+        if (nestedResults.length > 0) {
+          matchFound = true;
+        }
+      } else if (v?.toString()?.toLowerCase()?.includes(filter.toLowerCase())) {
+        matchFound = true;
+      }
+    });
+    if (!matchFound && d.antRecords) {
+      Object.keys(d?.antRecords).forEach((undername) => {
+        if (undername?.toLowerCase()?.includes(filter.toLowerCase())) {
+          matchFound = true;
+        }
+      });
+    }
+
+    if (matchFound) {
+      results.push(d);
+    }
+  });
+
+  return results;
+}
+
+const DomainsTable = ({
+  domainData,
+  loading,
+  filter,
+}: {
+  domainData: {
+    names: Record<string, AoArNSNameData>;
+    ants: Record<string, AoANTState>;
+  };
+  loading: boolean;
+  filter?: string;
+}) => {
+  const [{ walletAddress }] = useWalletState();
+  const [{ gateway }] = useGlobalState();
+  const [tableData, setTableData] = useState<Array<TableData>>([]);
+  const [filteredTableData, setFilteredTableData] = useState<TableData[]>([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (loading) {
+      setTableData([]);
+      setFilteredTableData([]);
+      return;
+    }
+    if (domainData) {
+      const newTableData: TableData[] = [];
+
+      Object.entries(domainData.names).map(([domain, record]) => {
+        const ant = domainData.ants[record.processId];
+        const data: TableData = {
+          openRow: <></>,
+          name: domain,
+          role:
+            getOwnershipStatus(
+              ant?.Owner,
+              ant?.Controllers,
+              walletAddress?.toString(),
+            ) ?? 'N/A',
+          processId: record.processId,
+          targetId: ant.Records?.['@']?.transactionId ?? 'N/A',
+          sourceCode: (ant as any)?.['Source-Code-TX-ID'],
+          undernames: {
+            used:
+              Object.keys(ant?.Records).filter((undername) => undername !== '@')
+                ?.length ?? 0,
+            supported: record.undernameLimit,
+          },
+          expiryDate: (record as any).endTimestamp ?? 'Indefinite',
+          status: isLeasedArNSRecord(record)
+            ? record.endTimestamp
+            : 'Indefinite',
+          action: <></>,
+          // metadata used for search and other purposes
+          antRecords: ant.Records,
+          domainRecord: record,
+        };
+        newTableData.push(data);
+      });
+
+      setTableData(newTableData);
+    }
+  }, [domainData, loading]);
+
+  useEffect(() => {
+    if (filter) {
```

Comment:
I think there may be something off here. If I put in a text that triggers the filter, then hit backspace repeatedly to clear it out, the filter updates and clears out. If instead I select the text and delete it or select and cut the text with cmd-x, the table does not update.